Utilities/rANS/test/test_ransCombinedIterator.cxx

```diff
@@ -0,0 +1,134 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   test_ransCombinedIterator.cxx
+/// @author michael.lettrich@cern.ch
+/// @since  2020-10-28
+/// @brief
+
+#define BOOST_TEST_MODULE Utility test
+#define BOOST_TEST_MAIN
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+
+#include "rANS/utils.h"
+
+struct test_CombninedIteratorFixture {
+  std::vector<uint16_t> a{0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0xe, 0xf};
+  std::vector<uint16_t> b{a.rbegin(), a.rend()};
+  size_t shift = 16;
+  std::vector<uint32_t> aAndB{0x0001000f, 0x0002000e, 0x0003000d, 0x0004000c, 0x0005000b,
+                              0x0006000a, 0x00070009, 0x00080008, 0x00090007, 0x000a0006,
+                              0x000b0005, 0x000c0004, 0x000d0003, 0x000e0002, 0x000f0001};
+};
+
+BOOST_FIXTURE_TEST_CASE(test_CombinedInputIteratorBase, test_CombninedIteratorFixture)
+{
+
+  auto readOP = [](auto iterA, auto iterB) -> uint32_t {
+    return *iterB + (*iterA << 16);
```

Comment:
should not be here ``*iterB + (uint32_t(*iterA)<<16`` to exclude overflow?

Reply:
Good point. Thanks for spotting it.